bin/jade.js

```diff
@@ -102,39 +103,24 @@ options.doctype = program.doctype;
 
 var files = program.args;
 
+// array of paths that are being watched
+
+var watchList = [];
+
 // compile files
 
 if (files.length) {
   console.log();
   if (options.watch) {
-    files.forEach(function(filename) {
-      function errorToString(e) {
-        return e.stack || /* istanbul ignore next */ (e.message || e);
-      }
-      try {
-        renderFile(filename);
-      } catch (e) {
-        // keep watching when error occured.
-        console.error(errorToString(e));
-      }
-      fs.watchFile(filename, {persistent: true, interval: 200},
-                   function (curr, prev) {
-        // istanbul ignore if
-        if (curr.mtime.getTime() === prev.mtime.getTime()) return;
-        try {
-          renderFile(filename);
-        } catch (e) {
-          // keep watching when error occured.
-          console.error(errorToString(e));
-        }
-      });
-    });
     process.on('SIGINT', function() {
       process.exit(1);
     });
-  } else {
-    files.forEach(renderFile);
+    // unwatch what's watched on exit
+    process.on("exit", function () {
+      watchList.forEach(fs.unwatchFile);
```

Comment:
I don't think we need to do this.  When node exits it will just stop watching the files at that point.